### ISSUE TITLE
Missing basic friction factor computation methods in the Navier Stokes module.

### ISSUE THREAD

**Issue by grmnptr** [idaholab/moose#24856]
## Reason
Simple (non-constant) friction factor correlations are needed for some porous flow configurations. Right now we don't have anything besides constants and linear correlations.

## Design
Add new materials.

## Impact
Will enable the application of the Navier Stokes module to a wider range of problems.

**Comment by GiudGiud** [comment]
We need a ParsedFunctorMaterial. I think we can allow a custom one for a classic dependence in the mean time

**Comment by grmnptr** [comment]
On long term I agree, but I have somebody who needs an exponential relationship on short term.

**Comment by joshuahansel** [comment]
@grmnptr Did you mean for your PR to close this? And thanks for the help!

**Comment by grmnptr** [comment]
No, I think there are plenty of other correlations which we are missing. I only added the `A*Re^B` relationship.

**Comment by GiudGiud** [comment]
@lindsayad has another PR that will satisfy part of this. I've been reluctant on adding it but now the justification is really weak

**Comment by GiudGiud** [comment]
then when we have a ParsedFunctorMaterial we will be golden here

**Comment by lindsayad** [comment]
Need reason for closing

**Comment by grmnptr** [comment]
Ultimately closed by:
https://github.com/idaholab/moose/pull/25242
